cfme/utils/appliance/__init__.py

```diff
@@ -2884,34 +2884,44 @@ def __init__(self):
         super(ApplianceStack, self).__init__()
 
     def push(self, obj):
-        was_before = self.top
+        stack_parent = self.top
```

Comment:
Other than the name change, I changed to using getattr() instead of ternary (getattr is safe against a None).

I also wrapped the hook method call in a general except block.